projects/graph/src/graph.py

```diff
@@ -2,8 +2,157 @@
 Simple graph implementation compatible with BokehGraph class.
 """
 
+import random
+
+class Queue:
+    def __init__(self):
+        self.queue = []
+    def enqueue(self, value):
+        self.queue.append(value)
+    def dequeue(self):
+        if (self.size()) > 0:
+            return self.queue.pop(0)
+        else:
+            return None
+    def size(self):
+        return len(self.queue)
+
+
+class Stack:
+    def __init__(self):
+        self.stack =[]
+    def push(self, value):
+        self.queue.append(value)
+    def pop(self):
+        if (self.size()) > 0:
+            return self.queue.pop(0)
+        else:
+            return None
+    def size(self):
+        return len
+
+"""Represent a graph as a dictionary of vertices mapping labels to edges."""
 
 class Graph:
-    """Represent a graph as a dictionary of vertices mapping labels to edges."""
     def __init__(self):
-        pass  # TODO
+        """create an empty dictionary for the vertices"""
+        self.vertices = {}
+
+    def add_vertex(self, vertex_id):
+        """add a vertex to the graph"""
+        self.vertices[vertex_id] = Vertex(vertex_id)
+
+    def add_edge(self, v1, v2):
+        """add an undirected edge to the graph"""
+        if v1 in self.vertices or v2 in self.vertices:
```

Comment:
In undirected graph both v1 and v2 need to be in self.vertices, so the 'or' should be 'and'.